pkg/fieldpath/fieldpath.go

```diff
@@ -42,6 +43,20 @@ func ExtractFieldPathAsString(obj interface{}, fieldPath string) (string, error)
 		return "", nil
 	}
 
+	annotationFieldPathParser, err := apiutil.NewAnnotationFieldPathParser()
+	if err != nil {
+		return "", nil
+	}
+
+	if annotationFieldPathParser.Validate(fieldPath) {
```

Comment:
Do we need to validate here given that we did this in `pkg/api/v1/conversion.go`?

Reply:
Yes, because we "validate" here just to check whether the fieldpath we got here is an annotation fieldpath or just an another kind of fieldpath. Unfortunately, we cannot just add this as a case for `switch` below.

Reply:
If so, I feel like it'd be better to have `Validate` to return `(key, bool)` to avoid always calling `ParseKey` twice.